lib/brainstem/concerns/controller_dsl.rb

```diff
@@ -233,6 +238,41 @@ def title(text, options = { nodoc: false })
           configuration[brainstem_params_context][:title] = \
             options.merge(info: text)
         end
+
+
+        def format_param_options(type = nil, options = {})
+          options = type if type.is_a?(Hash) && options.empty?
+
+          options[:type] = sanitize_param_data_type(type)
+          options[:item_type] = options[:item_type].to_s if options.has_key?(:item_type)
+          DEFAULT_PARAM_OPTIONS.merge(options)
+        end
+
+        DEFAULT_PARAM_OPTIONS = { nodoc: false, required: false }
+        private_constant :DEFAULT_PARAM_OPTIONS
+
+
+        def sanitize_param_data_type(type)
+          if type.is_a?(Hash) || type.blank?
+            deprecated_type_warning
+            type = DEFAULT_DATA_TYPE
+          end
+
+          type.to_s
+        end
+
+        DEFAULT_DATA_TYPE = 'string'
+        private_constant :DEFAULT_DATA_TYPE
+
+
+        def deprecated_type_warning
+          ActiveSupport::Deprecation.warn(
+            'Please specify the `type` of the parameter as the second argument. If not specified, '\
+              'it will default to `:string`. This default behavior will be deprecated in the next major '\
+              'version and will need to be explicitly specified. e.g. `post.valid :message, :text, required: true`',
```

Comment:
Heredoc might be easier for a string like this. Typing out all those backslashes is a pain, man!